src/casp/services/db/models.py

```diff
@@ -21,6 +21,17 @@ def __repr__(self):
         return self.email
 
 
+class UserActivity(db.Base):
+    id = sa.Column(sa.Integer, primary_key=True)
+    user_id = sa.Column(sa.Integer, sa.ForeignKey(f"{User.__tablename__}.id"))
+    cell_count = sa.Column(sa.Integer, default=0, nullable=False)
+    model_name = sa.Column(sa.String(255), nullable=False)
+    method = sa.Column(sa.String(255), nullable=True)
+    finished_time = sa.Column(sa.DateTime, default=datetime.datetime.now(datetime.timezone.utc))
+
+    __tablename__ = "user_activity"
+
+
```

Comment:
To get you started, you can add something at the end of the file that looks like:
```
# Add properties
sa.inspect(User).add_property(
    key="total_cells_processed",
    prop=column_property(
        sa.select(sa.func.sum(UserActivity.cell_count))
        .where(UserActivity.user_id == User.id)
        .correlate_except(UserActivity) -- might not need this line
        .scalar_subquery()
    )
)
```

and that will add a calculated field named `total_cells_processed`.  You then just add that field to the `UserAdminView.column_list` array in views.py

Let me know if that works!

Reply:
This is awesome, thank you!